Tinyxml2Serializer.cpp

```diff
@@ -1971,292 +1970,252 @@ bool CTinyxml2Deserializer::DeserializeAnalogSettings(std::vector<SAnalogDevice>
+{
+    pDataAvailable = false;
 
-    //    if (oXML.FindChildElem("Origin"))
-    //    {
-    //        oXML.IntoElem();
-    //        if (oXML.FindChildElem("X"))
-    //        {
-    //            sForcePlate.sOrigin.fX = (float)atof(oXML.GetChildData().c_str());
-    //        }
-    //        if (oXML.FindChildElem("Y"))
-    //        {
-    //            sForcePlate.sOrigin.fY = (float)atof(oXML.GetChildData().c_str());
-    //        }
-    //        if (oXML.FindChildElem("Z"))
-    //        {
-    //            sForcePlate.sOrigin.fZ = (float)atof(oXML.GetChildData().c_str());
-    //        }
-    //        oXML.OutOfElem();
-    //    }
+    pForceSettings.vsForcePlates.clear();
 
-    //    sForcePlate.vChannels.clear();
-    //    if (oXML.FindChildElem("Channels"))
-    //    {
-    //        oXML.IntoElem();
-    //        SForceChannel sForceChannel;
-    //        while (oXML.FindChildElem("Channel"))
-    //        {
-    //            oXML.IntoElem();
-    //            if (oXML.FindChildElem("Channel_No"))
-    //            {
-    //                sForceChannel.nChannelNumber = atoi(oXML.GetChildData().c_str());
-    //            }
-    //            if (oXML.FindChildElem("ConversionFactor"))
-    //            {
-    //                sForceChannel.fConversionFactor = (float)atof(oXML.GetChildData().c_str());
-    //            }
-    //            sForcePlate.vChannels.push_back(sForceChannel);
-    //            oXML.OutOfElem();
-    //        }
-    //        oXML.OutOfElem();
-    //    }
+    auto rootElem = oXML.RootElement();
+    if (!rootElem)
+    {
+        return true;
+    }
 
-    //    if (oXML.FindChildElem("Calibration_Matrix"))
-    //    {
-    //        oXML.IntoElem();
-    //        int nRow = 0;
+    auto forceElem = rootElem->FirstChildElement("Force");
+    if (!forceElem)
+    {
+        // No analog data available.
+        return true;
+    }
 
-    //        if (mnMajorVersion == 1 && mnMinorVersion < 12)
-    //        {
-    //            char strRow[16];
-    //            char strCol[16];
-    //            sprintf(strRow, "Row%d", nRow + 1);
-    //            while (oXML.FindChildElem(strRow))
-    //            {
-    //                oXML.IntoElem();
-    //                int nCol = 0;
-    //                sprintf(strCol, "Col%d", nCol + 1);
-    //                while (oXML.FindChildElem(strCol))
-    //                {
-    //                    sForcePlate.afCalibrationMatrix[nRow][nCol] = (float)atof(oXML.GetChildData().c_str());
-    //                    nCol++;
-    //                    sprintf(strCol, "Col%d", nCol + 1);
-    //                }
-    //                sForcePlate.nCalibrationMatrixColumns = nCol;
+    SForcePlate sForcePlate{};
+    sForcePlate.bValidCalibrationMatrix = false;
+    sForcePlate.nCalibrationMatrixRows = 6;
+    sForcePlate.nCalibrationMatrixColumns = 6;
```

Comment:
Isn't the column-count 8 now with the new data? Feel free to ignore if I'm incorrect or missing something.

Reply:
Theses are the previous values I don't know why they were chosen as the defaults. The max size is 12x12.